merkletree/proof_test.go

```diff
@@ -49,12 +49,18 @@ func authPathHash(ap *AuthenticationPath) []byte {
 }
 
 func verifyProof(t *testing.T, ap *AuthenticationPath, treeHash []byte, key string) {
+	// step 1. Verify the auth path of the returned leaf
 	hash := authPathHash(ap)
 	if !bytes.Equal(treeHash, hash) {
 		t.Fatal("Invalid proof")
 	}
 
-	// step 1. Verify the auth path of the returned leaf
+	// step 2. vrf_verify(i, alice) == true
+	// we probably want to use vrf.Verify() here instead
+	if !bytes.Equal(vrf.Compute([]byte(key), vrfPrivKey1), ap.LookupIndex()) {
```

Comment:
OK, I though this would just mean changing above line to: 
`if !vrf.Verify(vrf.Public(vrfPrivKey1), []byte(key), ap.LookupIndex(),  ap.VrfProof())`
but then the tests fail. So maybe this is for another PR...